tools/api-markdown-documenter/src/utilities/test/ApiItemUtilities.test.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ReleaseTag, type ApiItem } from "@microsoft/api-extractor-model";
+import { expect } from "chai";
+
+import { getEffectiveReleaseLevel } from "../ApiItemUtilities.js";
+
+describe("ApiItemUtilities", () => {
+	describe("getEffectiveReleaseTag", () => {
+		it("Item is tagged", () => {
+			const item = {
+				releaseTag: ReleaseTag.Alpha,
+			} as unknown as ApiItem;
+
+			expect(getEffectiveReleaseLevel(item)).to.equal(ReleaseTag.Alpha);
+		});
+
+		it("Tag is inherited", () => {
+			const parent = {
+				releaseTag: ReleaseTag.Beta,
+			} as unknown as ApiItem;
+
+			const item = {
+				parent,
+				releaseTag: undefined,
+			} as unknown as ApiItem;
+
+			expect(getEffectiveReleaseLevel(item)).to.equal(ReleaseTag.Beta);
+		});
+
+		it("Tag is more restrictive than ancestors", () => {
```

Comment:
I get the logic, but curious if this applies to FF in practice? We can only apply these tags to certain things (e.g. not properties) so do we hit this scenario at all?

Reply:
In practice, we should not hit this. But from an internal correctness perspective, it occurred to me while refactoring that the current logic was flawed.